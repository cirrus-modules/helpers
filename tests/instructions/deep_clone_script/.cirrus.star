load("cirrus", environ="env")
load("../../../lib.star", "task", "container", "deep_clone_script")

def main():
    return [
        _task_with_pr(),
        _task_without_pr()
    ]

def _task_with_pr():
    return task("deep_clone_script with PR", container(), instructions=[
        deep_clone_script()
    ])

def _task_without_pr():
    env = dict(environ)
    env.pop("CIRRUS_PR", None)

    return task("deep_clone_script without PR", container(), instructions=[
        deep_clone_script(env=env)
    ])
