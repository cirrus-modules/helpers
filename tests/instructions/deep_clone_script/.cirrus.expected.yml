task:
  name: deep_clone_script with PR
  container:
    cpu: 2
    memory: 4096
  clone_script:
    - git clone --recursive https://x-access-token:${CIRRUS_REPO_CLONE_TOKEN}@github.com/${CIRRUS_REPO_FULL_NAME}.git ${CIRRUS_WORKING_DIR}
    - git fetch origin pull/${CIRRUS_PR}/head:pull/${CIRRUS_PR}
    - git reset --hard ${CIRRUS_CHANGE_IN_REPO}
task:
  name: deep_clone_script without PR
  container:
    cpu: 2
    memory: 4096
  clone_script:
    - git clone --recursive --branch=${CIRRUS_BRANCH} https://x-access-token:${CIRRUS_REPO_CLONE_TOKEN}@github.com/${CIRRUS_REPO_FULL_NAME}.git ${CIRRUS_WORKING_DIR}
    - git reset --hard ${CIRRUS_CHANGE_IN_REPO}
